multimodal/src/autogluon/multimodal/models/mmdet_image.py

```diff
@@ -54,119 +55,171 @@ def __init__(
-            checkpoint = load_checkpoint(self.model, checkpoint, map_location="cpu")
+        if self.pretrained and checkpoint_file is not None:  # TODO: enable training from scratch
+            self.checkpoint = load_checkpoint(self.model, checkpoint_file, map_location="cpu")
 
+        # save the config and classes in the model for convenience
+        self.model.cfg = self.config
         if self.classes:
             self.model.CLASSES = self.classes
         else:
-            if num_classes == 20:  # TODO: remove hardcode
+            if self.checkpoint and "CLASSES" in self.checkpoint.get("meta", {}):
                 warnings.simplefilter("once")
                 warnings.warn(
-                    f"Using VOC classes because num_classes = {num_classes}. Provide data while init MultiModalPredictor if this is not VOC."
+                    f"Using classes provided in checkpoints: {self.checkpoint['meta']['CLASSES']}. Provide data while init MultiModalPredictor if this is not expected."
                 )
-                self.model.CLASSES = get_classes("voc")
-            elif num_classes == 80:
-                warnings.simplefilter("once")
-                warnings.warn(
-                    f"Using COCO classes because num_classes = {num_classes}. Provide data while init MultiModalPredictor if this is not COCO."
-                )
-                self.model.CLASSES = get_classes("coco")
-            elif "CLASSES" in checkpoint.get("meta", {}):
-                warnings.simplefilter("once")
-                warnings.warn(
-                    f"Using classes provided in checkpoints: {checkpoint['meta']['CLASSES']}. Provide data while init MultiModalPredictor if this is not expected."
-                )
-                self.model.CLASSES = checkpoint["meta"]["CLASSES"]
+                self.model.CLASSES = self.checkpoint["meta"]["CLASSES"]
             else:
                 raise ValueError("Classes need to be specified.")
 
-        self.model.cfg = self.config  # save the config in the model for convenience
-
-        self.prefix = prefix
-
         self.name_to_id = self.get_layer_ids()
         self.head_layer_names = [n for n, layer_id in self.name_to_id.items() if layer_id == 0]
 
-    def _load_checkpoint_and_config(self, checkpoint_name=None):
+    def dump_weights_and_config(self, save_path: str = "./", tokenizers: Optional[dict] = None):
+
+        weights_save_path = os.path.join(save_path, "model.pth")
+        configs_save_path = os.path.join(save_path, "config.py")
+
+        self._save_weights(save_path=weights_save_path)
+        self._save_configs(save_path=configs_save_path)
+
+        return save_path
+
+    def _save_weights(self, save_path=None):
+        if not save_path:
+            save_path = f"./{self.checkpoint_name}_autogluon.pth"
+
+        torch.save({"state_dict": self.model.state_dict(), "meta": {"CLASSES": self.model.CLASSES}}, save_path)
```

Comment:
Is it better to put `CLASSES ` in the config file? The hf_text and timm_image checkpoints contain only the model weights.

Reply:
We sometimes still load from MMDetection's model zoo and the `CLASSES` are in its checkpoints. To have a consistent loading operation, we'd better keep `CLASSES` in checkpoints.